tests/unittests/tests-nanocoap/tests-nanocoap.c

```diff
@@ -155,6 +155,10 @@ static void test_nanocoap__put_req(void)
     TEST_ASSERT_EQUAL_INT(fmt_opt_len, len);
     TEST_ASSERT_EQUAL_INT(COAP_FORMAT_TEXT, coap_get_content_type(&pkt));
 
+    len = coap_opt_add_uint(&pkt, COAP_OPT_ACCEPT, COAP_FORMAT_CBOR);
+    TEST_ASSERT_EQUAL_INT(fmt_opt_len, len);
+    TEST_ASSERT_EQUAL_INT(COAP_FORMAT_CBOR, coap_get_accept(&pkt));
+
     len = coap_opt_finish(&pkt, COAP_OPT_FINISH_PAYLOAD);
     TEST_ASSERT_EQUAL_INT(total_hdr_len + uri_opt_len + fmt_opt_len + 1, len);
```

Comment:
Also, this check needs to be amended:
```suggestion
    TEST_ASSERT_EQUAL_INT(
        total_hdr_len + uri_opt_len + fmt_opt_len + accept_opt_len + 1, len
    );
```